src/binary-reader-ir.cc

```diff
@@ -868,36 +865,36 @@ Result BinaryReaderIR::OnLocalNameLocalCount(Index index, Index count) {
 }
 
 Result BinaryReaderIR::OnInitExprF32ConstExpr(Index index, uint32_t value) {
-  Expr* expr = Expr::CreateConst(Const(Const::F32(), value, GetLocation()));
+  auto expr = new ConstExpr(Const(Const::F32(), value, GetLocation()));
```

Comment:
This particular case should probably be a `unique_ptr`. You're right that this should be cleaned up, but I've been trying to keep these changes smallish so they're easier to review.